nixos/modules/config/system-environment.nix

```diff
@@ -23,7 +23,7 @@ in
         strings.  The latter is concatenated, interspersed with colon
         characters.
       '';
-      type = types.attrsOf (types.loeOf types.str);
+      type = with types; attrsOf (either str (listOf str));
```

Comment:
No, that's not the problem. In both modules, the variables are lists of strings.